index.js

```diff
@@ -35,7 +35,9 @@ module.exports = (filename, options) => {
 		if (file.isNull() && file.stat && file.stat.isDirectory && file.stat.isDirectory()) {
 			zip.addEmptyDirectory(pathname, {
 				mtime: options.modifiedTime || file.stat.mtime || new Date(),
-				mode: file.stat.mode
+				// Set executable bit on directories if any other bits are set for that user/group/all
+				// Fixes creating unusable zip files on platforms that do not use an executable bit
+				mode: file.stat.mode | (((file.stat.mode >> 1) | (file.stat.mode >> 2)) & 0o111)
```

Comment:
Shouldn't this be fixed in https://github.com/gulpjs/vinyl-fs ? This plugin just passes the `.mode` it gets from there.

Reply:
Hmm, or actually, this seems like an issue in https://github.com/thejoshwolfe/yazl ? Since it should correctly handle a normal mode and do the right thing depending on the platform.

Reply:
Perhaps?  YAZL is already exposing `addEmptyDirectory` as something special - empty directories in zip files are only cross-platform if you use a default mode of 0, as far as I understand it, so maybe should just always passing `0` here would be fine?

Reply:
I think it would be very weird for YAZL to do anything with the 'mode' bits passed to it other than store it directly in the zip file (for people using zips on a single environment that's sometimes important).  Presumably most people are using `gulp-zip` as a part of build and distribution pipelines, where the output is expected to be cross-platform / just work.

Reply:
I just feel that YAZL should not allow creating ZIPs that end up being invalid on certain platforms. It would be better if YAZL either threw and error or did the right thing. Don't you agree?

As for fixing this here, I'm fine with just passing `0`.

---

> where the output is expected to be cross-platform / just work.

I cannot imagine any real scenario where you wouldn't want it to be cross-platform. You never know where the ZIP will eventually be used.

Reply:
The exception is when you're doing system maintenance kind of stuff - if I make a zip of my user directory, I would, perhaps, expect it to save the platform-specific mode bits and restore them when I unzip them.  But, since it doesn't include other platform-specific stuff like owner and group or security ACLs, it'd be a horrible choice for anything maintenance related, so maybe you're right and it just shouldn't be that way, but that seems to be the way zip is.  If you make a zip on Windows with the built-in PowerShell zip function, it *also* makes incompatible zip files in the same way :(.  This really seems like a bug in the unzipping side, but that's quite far out of our control =).

Upon consideration, I agree my earlier arguments about cross-platform should apply to YAZL, even if I also think that would be weird - I'll send the same PR setting the executable bits inside of `addEmptyDirectory()` to YAZL and see how that maintainer feels =)

Reply:
Sent a [PR](https://github.com/thejoshwolfe/yazl/pull/59) over to YAZL, but it's been silent so far.  Sounds like YAZL is [only semi-maintained](https://github.com/thejoshwolfe/yazl/issues/58), so might take some time before it gets resolved or rejected.